ts-client/src/ValorClient.ts

```diff
@@ -0,0 +1,163 @@
+import axios, { AxiosInstance } from 'axios';
+
+type Dataset = {
+  id: number;
+  name: string;
+  metadata: object;
+};
+
+type Model = {
+  id: number;
+  name: string;
+  metadata: object;
+};
+
+type TaskType =
+  | 'skip'
+  | 'empty'
+  | 'classification'
+  | 'object-detection'
+  | 'semantic-segmentation'
+  | 'embedding';
+
+type Evaluation = {
+  id: number;
+  model_name: string;
+  datum_filter: Dataset;
+  parameters: { task_type: TaskType; any };
+  status: 'pending' | 'running' | 'done' | 'failed' | 'deleting';
+  metrics: object[];
+  confusion_matrices: object[];
+};
+
+const metadataDictToString = (input: { [key: string]: string | number }): string => {
+  const result: { [key: string]: Array<{ value: string | number; operator: string }> } =
+    {};
+
+  Object.entries(input).forEach(([key, value]) => {
+    result[key] = [{ value: value, operator: '==' }];
+  });
+
+  return JSON.stringify(result);
+};
+
+export class ValorClient {
+  private client: AxiosInstance;
+
+  constructor(baseURL: string) {
+    this.client = axios.create({
+      baseURL,
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    });
+  }
+
+  public async getDatasets(queryParams: object): Promise<Dataset[]> {
+    const response = await this.client.get('/datasets', { params: queryParams });
```

Comment:
nit: is it worth adding docstrings to these methods (similar to the ones we use in our Python client)?

Reply:
yea i think so-- i just added